stream.c

```diff
@@ -102,6 +131,14 @@ void stream_handler(struct flow *f, uint32_t events)
 
         if (events & EPOLLOUT)
                 do {
+#ifdef WITH_TCPDIRECT
+                        if (t->opts->tcpd_gpu_pci_addr) {
+                                n = tcpdirect_send(fd, mbuf, opts->buffer_size, opts->send_flags);
```

Comment:
I'm unsure tcpdirect_send & tcpdirect_udma_send need to be different functions. I'm thinking tcpdirect_send() probably should deal only with dma-buf, and it's transparent to it whether it's udmabuf or cuda. But maybe more looking will show me why.